go-controller/pkg/ovn/base_secondary_layer2_network_controller.go

```diff
@@ -298,22 +297,9 @@ func (oc *BaseSecondaryLayer2NetworkController) InitializeLogicalSwitch(switchNa
 		return nil, fmt.Errorf("failed to create logical switch %+v: %v", logicalSwitch, err)
 	}
 
-	if err = oc.lsManager.AddSwitch(switchName, hostSubnets); err != nil {
+	if err = oc.lsManager.AddSwitch(switchName, hostSubnets, excludeSubnets...); err != nil {
```

Comment:
But that's all the commit is about and the commit message just talks about that:
```
Pre-allocate subnets in subnet ip allocator

Add the ability to pre-allocate subnets to the subnet ip allocator which
might be a common need.

Note that this is just a pre-allocation. The subnet ip allocator does
not distinguish between pre-allocated IPs vs normally allocated IPs and
will return the same ErrAllocated error on re-allocation.

In the process, refactor GetIPFullMask to be based on net.IPMask and
avoided parsing in a few places.
```

Perhaps I did not explain myself well but the whole intention of the commit and the commit message is this specific change.

Reply:
yeah you wrote "might be a common need", i think it would be clearer to give the context that you can provide excludedSubnets via NAD (or wherever we allow it) and it is common for localnet because the network is a shared external network where probably a bunch of the address space is already in use

Reply:
Reworded the commit message.